aten/src/ATen/SparseTensorImpl.cpp

```diff
@@ -80,7 +80,7 @@ int64_t SparseTensorImpl::storage_offset() const {
 }
 void SparseTensorImpl::set_indices_and_values_unsafe(const Tensor& indices, const Tensor& values) {
   AT_CHECK(allow_tensor_metadata_change(), "set_indices_and_values_unsafe is not allowed on Tensor created from .data or .detach()");
-  AT_ASSERT(!indices.is_variable() && !values.is_variable());  // They should be plain tensors!
+  AT_ASSERT(!indices.is_variable() && !values.is_variable());  // They should be plain tensors!  // TODO: change this to `.requires_grad()` when Variable and Tensor are merged
```

Comment:
I'm not sure we should remove this per se. An alternate claim we might make here, is that indices and values should not require grad (i.e., they should either themselves have requires_grad=False, or NoGrad guard is in effect.) The point is that this function doesn't work with autograd recording things, and so passing it things like that is a bug and we should report it.

Reply:
Would it be a good idea to check `torch::NoGradGuard` in `requires_grad()` so that it can be the function that handles both cases?